ghcide/src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -1553,7 +1551,7 @@ rangesForBinding' b (L l (IEThingWith _ thing _  inners labels))
 rangesForBinding' _ _ = []
 
 -- | 'matchRegex' combined with 'unifySpaces'
-matchRegexUnifySpaces :: T.Text -> T.Text -> Maybe [T.Text]
+matchRegexUnifySpaces :: T.Text -> T.Text -> Maybe (NonEmpty T.Text)
```

Comment:
I get why this is the return type, but it's funny to see `Maybe (NonEmpty a)` instead of `[a]`!

Reply:
It *is* funny, but in my experience it's actually the more convenient/"right" choice, whenever you want to do something special for the empty list case.